libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL-fp.cpp

```diff
@@ -0,0 +1,36 @@
+
+#include <WiFiClientSecureBearSSL.h>
+
+namespace BearSSL {
+
+static uint8_t htoi (unsigned char c)
+{
+    return c >= '0' && c <= '9'? c - '0'
+         : c >= 'A' && c <= 'F'? c - 'A' + 10
+         : c >= 'a' && c <= 'f'? c - 'a' + 10
+         : 255;
+}
+
+void WiFiClientSecure::setFingerprint(const String& fingerprint) {
+
+    uint8_t fp [20];
+    uint8_t c, d;
+    int idx = 0;
+    const char* _fingerprint = fingerprint.c_str();
+
+    while (idx < 20 && (c = *_fingerprint++) && (d = *_fingerprint++)) {
```

Comment:
Assuming we're sscanf'ing manually here, can we use `c=pgm_read_byte(_fingerprint++) && d=pgm_read_byte...` instead?  Everything else in BearSSL silently "just works" with PROGMEM strings/arrays (except for `write()` which I really should have just use `memcpy_p` and be done with it).

Reply:
I don't think we can always assume fingerprint is in progmem. We would have to have two functions{,_P}. String is already here, takes care of progmem, and is released when we are done.

Reply:
As below, no assumption is needed.  RAM and flash can be read w/pgm_read*.  It's slower than plain pointer reads, true, but when compared to the actual SSL connection it's absolutely not noticeable (and this is only run one time before connection, not in a loop for every step of the process).